### DISCUSSION TITLE
Unhandled extension util/ui when using  <TargetFramework> in Directory.Build.props

### DISCUSSION THREAD

**Discussion by malyovanik** [wixtoolset/issues#7766] [Questions] [answered]
I migrated my project from v3 to v4 and found one issue. My extensions are unresolved, in my project. I discovered the reason is using the <TargetFramework> tag in the Directory.Build.props file.  When I comment or delete this line - all works fine, but this option needs me for other projects. In v3 all was good in v4 - nope. Please help ))))

**Comment by YevheniiSych** [comment]
I have the same issue!

**Comment by robmen** [comment, marked as answer]
The `TargetFramework` property is highly impactful. Setting it incorrectly is highly disruptive to project restore and build. Condition the `TargetFramework` property, so it only applies to the projects where it is appropriate. Maybe something like ` Condition=" '$(MSBuildProjectExtension)'=='.csproj' "` is appropriate.

**Reply by malyovanik** [reply]
Thanks for your reply. I Tried many options, and now I will try to do it again like you posted. I understand I can't use TargetFramework for Wix projects right??

**Reply by robmen** [reply]
What would it mean to use `TargetFramework` for WiX projects?

**Reply by malyovanik** [reply]
I mean, If I set this property for all projects in my directories it will not work for wix projects?

**Reply by malyovanik** [reply]
My configuration of projects worked fine for Wix v3, and when migrated Wix projects to v4 I found this issue.

**Reply by robmen** [reply]
No, it works in v4. Setting `TargetFramework` in `Directory.Build.props` will change the `TargetFramework` of the .wixprojs that are below that `Directory.Build.props`. The issue is that if you set it to a `TargetFramework` to an invalid value for projects, those projects break. So, it depends on the value of `TargetFramework`. In this case, you've apparently picked one that breaks your .wixprojs.

**Reply by robmen** [reply]
WiX v3 doesn't support `TargetFramework` because it was implemented before `TargetFramework` existed. As a result, WiX v3 cannot resolve modern C# projects and does not interact with NuGet fully, and other issues I'm not remembering off the top of my head.

`TargetFramework` is completely broken in WiX v3. It does not work there.

**Reply by malyovanik** [reply]
ohh, I see. Thanks. Now it`s clear to me, why I have this issue and why it works in v3. Thank you again for your reply. I will resolve this thread when find out how to fix it for me.

**Reply by malyovanik** [reply]
@robmen  Thank you a lot. It helps me. If someone has a question about that, ask me)))
